### ISSUE TITLE
sometimes when scrolling, scroll position jumps to random location

### ISSUE THREAD

**Issue by philipag** [amit115532/SmoothScrolling#3]
Probably some sort of reentrancy issue.   Unfortunately I cannot say exactly when it happens because it happens only a couple of time a day.

**Comment by amit115532** [comment]
Yeah, it does happen sometimes. Not really sure why, I think it happens when the framerate drops dramatically (like when visual studio is doing some heavy operations).
Ill try to fix it soon
